pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -474,39 +477,89 @@ public boolean treeIsEmpty() {
      * Returns an iteration over the reported violations.
      *
      * @return an iterator
+     *
+     * @deprecated The {@link ReportTree} is deprecated
      */
+    @Deprecated
     public Iterator<RuleViolation> treeIterator() {
         return violationTree.iterator();
     }
 
+    /**
+     * @deprecated Use {@link #getViolations()}
+     */
+    @Deprecated
     @Override
     public Iterator<RuleViolation> iterator() {
         return violations.iterator();
     }
 
+
+    /**
+     * Returns an unmodifiable list of violations that were suppressed.
+     */
+    public List<SuppressedViolation> getSuppressedRuleViolations() {
+        return Collections.unmodifiableList(suppressedRuleViolations);
+    }
+
+    /**
+     * Returns an unmodifiable list of violations that have been
+     * recorded until now. None of those violations were suppressed.
+     */
+    public List<RuleViolation> getViolations() {
+        return Collections.unmodifiableList(violations);
+    }
+
+
+    /**
+     * Returns an unmodifiable list of processing errors that have been
+     * recorded until now.
+     */
+    public List<ProcessingError> getProcessingErrors() {
+        return Collections.unmodifiableList(errors);
+    }
+
+
+    /**
+     * Returns an unmodifiable list of configuration errors that have
+     * been recorded until now.
+     */
+    public List<ConfigurationError> getConfigErrors() {
```

Comment:
I think we should name this without abbreviation, since this is the only one, where we use abbrev.
I think, it should be called `getConfigurationErrors()`.